docs/use_cases/advanced_retrieval_augmented_generation.md

```diff
@@ -0,0 +1,284 @@
+
+As for the data, we will fetch information from Vector Hub articles and convert the html files to documents.
+
+```python
+from haystack.components.fetchers import LinkContentFetcher
+from haystack.components.converters import HTMLToDocument
+
+link_fetcher = LinkContentFetcher()
+converter = HTMLToDocument()
+```
+One great thing about Haystack is that it offers a wide selection of pipeline components, including the LinkContentFetcher and HTMLToDocument modules, which are exactly what we need. Typically, components in Haystack are first defined, then added to a pipeline. We can build and configure our pipeline with these building blocks and they will all be executed once the pipeline is run. If we wanted to run our components individually, we could do that like this:
+
+```python
+# What would be better than fetching Vector Hub texts in a tutorial on Vector Hub?
+fetched = link_fetcher.run(urls=["<https://superlinked.com/vectorhub/>"])
+converted = converter.run(sources=fetched["streams"])
+```
+However, this is not advised outside of development, as pipelines offer a cleaner, more streamlined interface for executing our code.
+
+## Pre-retrieval
+
+### Document cleaning
+Data is the lifeblood of any Machine Learning (ML) model, and its quality directly impacts the performance of RAG systems. Cleaning involves removing noise such as irrelevant information, correcting typos, and standardizing formats to ensure the data is optimized for machine processing. Clean data not only improves the efficiency of retrieval and generation but also significantly enhances the quality of the generated text.
+
+For this tutorial, we will be using a simple document cleaner that removes empty lines and extra whitespaces. If we would be working with messier data, could remove repeated substrings, such as html tags or even write custom scripts to standardize words and correct typos. Your cleaning routine will depend heavily on your specific use case and goals.
+
+```python
+from haystack.components.preprocessors import DocumentCleaner
+
+cleaner = DocumentCleaner(
+	remove_empty_lines=True,
+	remove_extra_whitespaces=True,
+	remove_repeated_substrings=False)
+```
+
+### Chunking
+Chunking refers to the process of breaking down large pieces of text into more manageable, logically coherent units. This step is crucial for efficient retrieval, as it enables the system to focus on the most relevant segments of text when generating responses. Effective chunking strategies can dramatically improve the relevance and cohesion of the generated content.
+
+```python
+from haystack.components.preprocessors import DocumentSplitter
+
+splitter = DocumentSplitter(split_by="sentence", split_length=3, split_overlap=2)

```

Comment:
maybe we could link the chunking research article here as an argument for choosing sentences as the basis of chunks